### ISSUE TITLE
Why is the ACCESS_TOKEN necessary for sending a message?

### ISSUE THREAD

**Issue by axos88** [jgorset/facebook-messenger#259]
If the bot lives on a single page, the access token should be part of the configuration. 

If the bot lives on multiple pages, the configuration provides is able to provide the access token based on the page id.

Why is the access token necessary to be sent in each time (for non-responses)? Shouldn't the page id be enough for this (and only in case the bot lives on multiple pages)?

Can you share an example showing how to access the active configuration provider to fetch this info manually currently?

**Comment by jgorset** [comment]
Hi @axos88! That's a good question, and there may be some room here to improve the architecture.

In the event that you're wanting to send a user a message out of the blue (e.g. a message that is not in response to a message a user sent your page), then we'll have to pass that information somehow to `Bot.deliver`. Right now, that's done by passing `access_token`, but it'd be better to let developers simply pass something like `page_id` and figure out what the access token is for them! I'd be very happy to merge a pull request for this, so I'll tag this issue and leave it open in case someone wants to give it a go.

**Comment by jgorset** [comment]
>Can you share an example showing how to access the active configuration provider to fetch this info manually currently?

It depends on your choice of configuration provider, but the reference implementation defines [`access_token_for(*)`](https://github.com/jgorset/facebook-messenger/blob/master/lib/facebook/messenger/configuration/providers/base.rb#L30) that I intended for this. So, if you've written your own provider that gets your access token from a database, for example, I'd recommend something like this:

```
YourFacebookMessengerConfigurationProvider.access_token_for(page_id)
```

**Comment by axos88** [comment]
> > Can you share an example showing how to access the active configuration provider to fetch this info manually currently?
> 
> It depends on your choice of configuration provider, but the reference implementation defines [`access_token_for(*)`](https://github.com/jgorset/facebook-messenger/blob/master/lib/facebook/messenger/configuration/providers/base.rb#L30) that I intended for this. So, if you've written your own provider that gets your access token from a database, for example, I'd recommend something like this:
> 
> ```ruby
> YourFacebookMessengerConfigurationProvider.access_token_for(page_id)
> ```

I meant retrieving the active configuration provider without having to tightly copule to the actual implementation. Like `Bot.get_active_configuration_provider`, or something simliar, so that changing the configuration provider would not break the code.

**Comment by jgorset** [comment]
Ah, right! `Facebok::Messenger.config.provider` should do the trick.

Here's the implementation: https://github.com/jgorset/facebook-messenger/blob/master/lib/facebook/messenger.rb#L18

**Comment by axos88** [comment]
Thanks @jgorset , I'll look into this. I would argue that it would be great to add this info to the docs!

**Comment by jgorset** [comment]
Agreed! [Done](ba756c8)!

**Comment by jgorset** [comment]
I got around to refactor this and released it in version 2.0.0. Thanks for the idea!
